internal/httputil/httpapi.go

```diff
@@ -49,7 +55,10 @@ type BasicAuth struct {
 
 // MakeAuthAPI turns a util.JSONRequestHandler function into an http.Handler which authenticates the request.
 func MakeAuthAPI(
-	metricsName string, userAPI userapi.UserInternalAPI,
+	metricsName string,
+	userAPI userapi.UserInternalAPI,
+	userConsentCfg config.UserConsentOptions,
+	requireConsent bool,
```

Comment:
From some call sites in the client API it's not really clear what this bool does. Can we type & const it somewhere?
```
type Consent bool

const (
  ConsentRequired Consent = true
  ConsentNotRequired Consent = false
)
```